src/prompt/utils/message.ts

```diff
@@ -98,21 +98,50 @@ export class Msg {
     return { role: 'function', content: contentString, name };
   }
 
+  /** Create a function call message with argumets. */
+  static toolCalls(
+    tool_calls: Prompt.Msg.ToolCall[],
+    opts?: {
+      /** The name descriptor for the message.(message.name) */
+      name?: string;
```

Comment:
it looks like openai may have removed support for `name`, i'm guessing in favor of the stateful api for that use case

Reply:
Yeah, it does look gone. I also noticed they started naming the different message type [here](https://platform.openai.com/docs/api-reference/chat/create). We should probably stay in line with those to avoid confusion, but I still prefer the even more explicit assistant breakdown. What do you think of:
- System
- User
- Assistant
  - AssistantTool
  - AssistantFunction
- Tool
- Function

Reply:
Sounds good; just including a screenshot to make sure we're talking about the same naming.

I'm fine w/ the more explicit version. I was just following the pattern you had done for functions previously as best I could pattern-match.

<img width="644" alt="CleanShot 2023-11-07 at 15 41 31@2x" src="https://github.com/dexaai/dexter/assets/552829/53dbf847-bc87-4b38-aed7-6c4a6f8f1513">

Reply:
Also dealing w/ this which seems to be a typing bug in the latest `openai` release https://github.com/openai/openai-node/issues/462